pkg/gadgets/run/tracer/run.go

```diff
@@ -85,10 +93,34 @@ func (g *GadgetDesc) GetGadgetInfo(params *params.Params, args []string) (*types
 	}
 
 	ret := &types.GadgetInfo{
-		ProgContent: gadget.EbpfObject,
+		ProgContent:    gadget.EbpfObject,
+		GadgetMetadata: &types.GadgetMetadata{},
 	}
-	if err := yaml.Unmarshal(gadget.Metadata, &ret.GadgetDefinition); err != nil {
-		return nil, fmt.Errorf("unmarshaling definition: %w", err)
+
+	spec, err := loadSpec(ret.ProgContent)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(gadget.Metadata) == 0 {
```

Comment:
I would argue that if there is no metadata and the flag `validateMetadata` is `true` then we should print out an error and exit. But since `true` is the default value it wouldn't be that great